internal/util/prometheus.go

```diff
@@ -0,0 +1,55 @@
+package util
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+type ControllerFunctionalPrometheusMetrics struct {
+	// number of post /config to proxy successfully
+	ConfigPassCounter prometheus.Counter
+	// number of post /config to proxy failed
+	ConfigFailureCounter prometheus.Counter
+	// number of ingress analysis failure
+	ParseFailureCounter prometheus.Counter
+
+	// duration of last successful confiuration sync
+	ConfigureDurationHistogram prometheus.Histogram
+}
+
+func (ctrlMetrics *ControllerFunctionalPrometheusMetrics) GeneratePrometheusCounter(name, help string) prometheus.Counter {
+	return prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: name,
+			Help: help,
+		},
+	)
+}
+
+func (ctrlMetrics *ControllerFunctionalPrometheusMetrics) GeneratePrometheusHistogram(name, help string) prometheus.Histogram {
```

Comment:
Maybe `NewPrometheusHistogram` would be a better name? `Generate` implies data generation and I cannot see any happening here.

Reply:
Agree. 
NewPrometheusHistogram is aligned with Prometheus code package.